packages/ag-charts-core/src/options/commonOptionsDefs.ts

```diff
@@ -1,24 +1,62 @@
-import type { FillOptions, FontOptions, LineDashOptions, StrokeOptions } from 'ag-charts-types';
+import type { AgGradientFill, FillOptions, FontOptions, LineDashOptions, StrokeOptions } from 'ag-charts-types';
 
+import { isObject } from '../utils/typeGuards';
 import {
     type OptionsDefs,
+    and,
+    array,
     arrayOf,
+    constant,
     number,
     object,
+    optionsDefs,
     or,
     positiveNumber,
     ratio,
+    required,
     string,
     union,
 } from '../utils/validation';
 
+export const operationsDef: OptionsDefs<any> = {
+    $ref: string,
+    $path: string,
+    $if: array,
+    $or: array,
+    $and: array,
+    $eq: array,
+    $mul: array,
+    $round: array,
+    $rem: array,
+    $mix: array,
+    $mixEach: array,
+    $foregroundBackgroundMix: array,
+    $foregroundBackgroundAccentMix: array,
+};
+
+const operationKeys = Object.keys(operationsDef);
+// TODO: AG-13892 Attach a description to this validator
+export const operation = and(
+    (value: unknown) => isObject(value) && operationKeys.includes(Object.keys(value)[0]),
+    optionsDefs(operationsDef)
+);
+
+const gradientOptionsDef: OptionsDefs<AgGradientFill> = {
+    type: required(constant('gradient')),
+    direction: union('horizontal', 'vertical'),
+    colorStops: object,
+    bounds: union('series', 'item', 'axes'),
+};
+
+const gradient = optionsDefs(gradientOptionsDef, 'a gradient');
+
 export const fillOptionsDef: OptionsDefs<FillOptions> = {
-    fill: or(string, object),
+    fill: or(string, gradient, operation),
     fillOpacity: ratio,
 };
 
 export const strokeOptionsDef: OptionsDefs<StrokeOptions> = {
-    stroke: string,
+    stroke: or(string, operation),
```

Comment:
do we need to support gradient for stroke too? (example implies this)

Reply:
It's not part of the API, if it's in a current example it would be an error too since this previous validator would have blocked it.